### PR TITLE
Authorization re-write

### PR DESCRIPTION
**Pull request by Ceredron**
## Description
Re-wrote authorization to support delegations and system user. Also general clean-up.

Note: Do we need to do the instance check (i.e verify that a given onBehalfOf is sender/recipient of a file transfer) for legacy calls as we did before? Code currently does this so that legacy logic does not change

## Related Issue(s)
- #576 

## Verification
- [X] **Your** code builds clean without any errors or warnings
- [X] Manual testing done (required)
- [X] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [X] All tests run green

## Documentation
- [X] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

## Summary by CodeRabbit

- **New Features**
	- Introduced a new project, `Altinn.Broker.Common`, to enhance shared functionality.
	- Added extension methods for organization number validation and file transfer entity checks.

- **Bug Fixes**
	- Enhanced access validation logic across various handlers for file transfer operations.

- **Documentation**
	- Updated configuration to include local settings management.

- **Chores**
	- Removed unused `CallerIdentity` class and related token handling across multiple controllers and requests.

- **Tests**
	- Removed a test case related to file access permissions for consumers.

### CODE REVIEW

File: src/Altinn.Broker.Common/StringExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    /// <param name="orgNumber">The organization number to format</param>
+    /// <returns>Returns the last sequence succeeding a colon.</returns>
+    public static string WithoutPrefix(this string orgNumber)
+    {
+        if (string.IsNullOrWhiteSpace(orgNumber))
+        {
+            return string.Empty;
+        }
+        return orgNumber.Split(":").Last();
+    }
```
[the remark this answers is not in the document]
Comment (Ceredron) [reply src/Altinn.Broker.Common/StringExtensions.cs:30]: Split returns an empty array when original string is empty. This is guarded in the function.

File: src/Altinn.Broker.Core/StringExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    /// <param name="orgNumber">The organization number to format</param>
+    /// <returns>Returns the last sequence succeeding a colon.</returns>
+    public static string WithoutPrefix(this string orgNumber)
+    {
+        if (string.IsNullOrWhiteSpace(orgNumber))
+        {
+            return string.Empty;
+        }
+        return orgNumber.Split(":").Last();
+    }
```
[the remark this answers is not in the document]
Comment (Ceredron) [reply src/Altinn.Broker.Core/StringExtensions.cs]: So ugly and unreadable

**Comment by RagnarFatland-Avanade** [comment]
>Do we need to do the instance check (i.e verify that a given onBehalfOf is sender/recipient of a file transfer) for legacy calls as we did before? Code currently does this so that legacy logic does not change

Can confirm Altinn 2 Legacy solution had explicit check on specific sender/recipient (and not delgated parties) so calls through Legacy solution should reflect this.

**Review by Andreass2 — APPROVED** [review]
Looks good

File: src/Altinn.Broker.Common/ClaimsPrincipalExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            return orgClaim.Value.WithoutPrefix(); // Normalize to same format as elsewhere
+        }
+        var consumerClaim = user.Claims.FirstOrDefault(c => c.Type == "consumer");
+        if (consumerClaim is not null)
+        {
+            var consumerObject = JsonSerializer.Deserialize<TokenConsumer>(consumerClaim.Value);
+            return consumerObject.ID.WithoutPrefix();
+        }
+        return null;
+    }
```
[the remark this answers is not in the document]
Comment (Ceredron) [reply src/Altinn.Broker.Common/ClaimsPrincipalExtensions.cs:32]: Fail hard on malformed token

**Pull request merged**